src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -944,10 +964,10 @@ emit_vector64_vector128_t (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 		return ins;
 	}
 	case SN_get_Zero: {
-		return emit_simd_ins (cfg, klass, OP_XZERO, -1, -1);
+		return emit_xzero (cfg, klass);
 	}
 	case SN_get_AllBitsSet: {
-		MonoInst *ins = emit_simd_ins (cfg, klass, OP_XZERO, -1, -1);
+		MonoInst *ins = emit_xzero (cfg, klass);
 		return emit_xcompare (cfg, klass, etype->type, ins, ins);
```

Comment:
What is this one going to emit? I'd presume `xorps, cmpps` which is going to be inefficient.

This really should be directly emitting `vcmptrueps` (which is `vcmpps` with an immediate of `0x0F`).

Reply:
This will be the output for `AllBitsSet`:

```csharp
private static Vector128<T> Test<T>() where T: struct
    => Vector128<T>.AllBitsSet;
```

```asm
0000000000000000 <loWorld_Program_Test_double____>:
<BB>:1
   0:   c5 f9 76 c0             vpcmpeqd %xmm0,%xmm0,%xmm0
   4:   c5 fa 7f 07             vmovdqu %xmm0,(%rdi)
   8:   c3                      retq   
***
```

Reply:
Thanks for confirming, looks like that's probably optimized somewhere else then and is being done correctly.